src/averageAges.js

```diff
@@ -15,47 +15,62 @@
  * @return {number}
  */
 function calculateMenAverageAge(people, century) {
-  // write code here
-  // learn how to use array methods like .filter .map .some .every .find .reduce
-  // avoid using loop and forEach
-  // replace `if ()` statement with &&, || or ?:
-  // without nesting
+  let males = people.filter(({ sex }) => sex === 'm');
+
+  if (century) {
+    males = males.filter(male => Math.ceil(male.died / 100) === century);
+  }
+
+  const averageAges = males.map(
+    male => {
+      return male.died - male.born;
+    })
+    .reduce(
+      (ageOfPreviousMale, ageOfCurrentMale) =>
+        ageOfPreviousMale + ageOfCurrentMale
+    ) / males.length;
+
+  return averageAges;
 }
 
-/**
- * Implement calculateWomenAverageAge function
- *
- * Function returns average age of women in array. If `withChildren` is
- * specified then function calculates average age only for women with children
- *
- * Hint: To check if a woman has children you should find someone who mention
- * her as mother.
- *
- * @param {object[]} people
- * @param {boolean} withChildren - optional
- *
- * @return {number}
- */
 function calculateWomenAverageAge(people, withChildren) {
-  // write code here
+  let women = people.filter(({ sex }) => sex === 'f');
+
```

Comment:
Part of this line has code that is repeated. Could you possibly turn it into a variable or a global variable?

`const isFemale = (person) => person.sex === 'f';`